client/invokeEntrance.ftl

```diff
@@ -20,17 +15,106 @@
 [#-- Validate Command line options are right for the entrance --]
 [#assign validCommandLineOptions = getCompositeObject(entranceEntry.Configuration, getCommandLineOptions()) ]
 
-[#-- Find and invoke the Entrance Macro --]
-[#-- Entrances provided by explicit providers are preferred over the shared provider --]
-[@invokeEntranceMacro
-    type=entranceType
-/]
+[#-- Input seeding for handling invoking different passes through the entrance --]
+[#assign ENTRANCE_PASS_INPUT_SEEDER = "entrancepass" ]
 
-[#-- Write generated logs out to loggers --]
-[@writeLogs
-    writers=getCommandLineOptions().Logging.Writers
-/]
+[#assign entrancePasses = []]
+[#assign activePass = {} ]
+
+[#-- The entrance passes are defined during the generation contract process --]
+[#-- If a contract stage is passed to the engine then we treat those as the entrance passes required --]
+[#if ((getCommandLineOptions().Contract.Stage)!{})?has_content ]
+    [#assign entrancePasses = (getCommandLineOptions().Contract.Stage.Steps)?filter( x -> x.Type == "process_template_pass") ]
+[#else]
+
+    [#-- Configure the pass as a generation contract --]
+    [#assign entrancePasses = [
+        {
+            "Type" : "process_template_pass",
+            "Parameters" : {
+                "outputFormat" : "",
+                "outputType" : "contract",
+                "pass" : "generationcontract",
+                "deploymentUnitSubset" : "generationcontract",
+                "passAlternative" : "",
+                "outputFileName" : getCommandLineOptions().Output.FileName
+            }
+        }
+    ]]
+[/#if]
+
+[#list entrancePasses as entrancePass ]
+
+    [#-- set this globally to allow input seeding to access the pass during input refresh --]
+    [#assign activePass = entrancePass]
+
+    [@registerInputSeeder
```

Comment:
Move seeder registration/config outside loop and replace with a call to `[@scheduleInputStateRefresh /]`

Note that this needs  PR #1669 to be merged.